src/devices/machine/intelfsh.cpp

```diff
@@ -982,7 +998,68 @@ void intelfsh_device::write_full(uint32_t address, uint32_t data)
 			logerror( "unexpected %08x=%02x in FM_SETMASTER:\n", address, data & 0xff );
 			break;
 		}
-		m_flash_mode = FM_NORMAL;
+		m_flash_mode = FM_READSTATUS;
+		break;
+	case FM_WRITEBUFFER1:
+		// Datasheets don't specify what happens when the word count is outside of
+		// the valid range so clamp and pray
+		if ( m_bits == 16 )
+			m_write_buffer_count = (data > 0xf ? 0xf : data) + 1;
+		else
+			m_write_buffer_count = (data > 0x1f ? 0x1f : data) + 1;
```

Comment:
Please use `std::min<uint32_t>(data, ...)` rather than the ternary conditional operator to express this.